topology/pynest/tests/test_rotated_rect_mask.py

```diff
@@ -0,0 +1,406 @@
+        mask = topo.CreateMask('box', maskdict)
+        gid_list = topo.SelectNodesByMask(layer, cntr, mask)
+        sorted_gid_list = sorted(gid_list)
+
+        self.assertEqual(sorted_gid_list, [33, 38, 43, 59, 64, 69, 85, 90, 95])
+
+        # Test with a polar angle of 135 degrees. This should be opposite of
+        # the ones obtained by a polar angle of 45 degrees.
+        maskdict = {'lower_left': [-0.5, -1.5, -1.5],
+                    'upper_right': [0.5, 1.5, 1.5],
+                    'polar_angle': 135.}
+        mask = topo.CreateMask('box', maskdict)
+        gid_list = topo.SelectNodesByMask(layer, cntr, mask)
+        sorted_gid_list = sorted(gid_list)
+
+        self.assertEqual(sorted_gid_list, [35, 40, 45, 59, 64, 69, 83, 88, 93])
+
+        # Test two symmetric masks in x and z direction. One with no polar
+        # angle and one with a polar angle of 90 degrees. As the masks are
+        # symmetrical in  x and z, a polar angle of 90 degrees should give the
+        # same GIDs as the one without a polar angle.
+        maskdict = {'lower_left': [-1., -0.5, -1.],
+                    'upper_right': [1., 0.5, 1.]}
+        mask = topo.CreateMask('box', maskdict)
+        gid_list = topo.SelectNodesByMask(layer, cntr, mask)
+        sorted_gid_list_1 = sorted(gid_list)
+
+        self.assertEqual(sorted_gid_list_1,
+                         [38, 39, 40, 63, 64, 65, 88, 89, 90])
+
+        maskdict = {'lower_left': [-1., -0.5, -1.],
+                    'upper_right': [1., 0.5, 1.],
+                    'polar_angle': 90.}
+        mask = topo.CreateMask('box', maskdict)
+        gid_list = topo.SelectNodesByMask(layer, cntr, mask)
+        sorted_gid_list = sorted(gid_list)
+
+        self.assertEqual(sorted_gid_list, [38, 39, 40, 63, 64, 65, 88, 89, 90])
+
+        self.assertEqual(sorted_gid_list_1, sorted_gid_list)
+
+    def test_RotatedBoxMaskByAzimuthAndPolarAngle(self):
+        """Test rotated box mask with azimuth and polar angle."""
+
+        pos = [[x*1., y*1., z*1.] for x in range(-2, 3)
+               for y in range(-2, 3)
+               for z in range(-2, 3)]
+
+        layer = topo.CreateLayer({'positions': pos, 'extent': [5., 5., 5.],
+                                  'elements': 'iaf_psc_alpha'})
+
+        # Test with a azimuth angle and polar angle of 45 degrees.
+        maskdict = {'lower_left': [-0.5, -1.5, -1.5],
+                    'upper_right': [0.5, 1.5, 1.5],
+                    'azimuth_angle': 45.,
+                    'polar_angle': 45.}
+        mask = topo.CreateMask('box', maskdict)
+        cntr = [0., 0., 0.]
+        gid_list = topo.SelectNodesByMask(layer, cntr, mask)
+        sorted_gid_list = sorted(gid_list)
+
+        self.assertEqual(sorted_gid_list,
+                         [38, 39, 44, 58, 59, 64, 69, 70, 84, 89, 90])
+
+    def test_RotatedRectangleOutsideOrigo(self):
+        """Test rotated rectangle where mask is outside origo."""
```

Comment:
As it is the mask that contains things, not origo, perhaps something along the lines of "where origo is not in mask"?